datadog_checks_base/datadog_checks/checks/base.py

```diff
@@ -89,17 +90,31 @@ def __init__(self, *args, **kwargs):
                 False,
                 "DEPRECATION NOTICE: `in_developer_mode` is deprecated, please stop using it.",
             ],
+            'no_proxy': [
+                False,
+                "DEPRECATION NOTICE: The `no_proxy` config option has been renamed "
+                "to `skip_proxy` and will be removed in a future release.",
+            ],
         }
 
     @property
     def in_developer_mode(self):
         self._log_deprecation('in_developer_mode')
         return False
 
-    def get_instance_proxy(self, instance, uri):
-        proxies = self.proxies.copy()
+    def get_instance_proxy(self, instance, uri, proxies=None):
+        proxies = proxies if proxies is not None else self.proxies.copy()
+        proxies['no'] = os.getenv('no_proxy', os.getenv('NO_PROXY', None))
+
+        deprecated_skip = instance.get('no_proxy', None)
+        skip = (
+            is_affirmative(instance.get('skip_proxy', not self._use_agent_proxy)) or
+            is_affirmative(deprecated_skip)
+        )
+
+        if deprecated_skip is not None:
+            self._log_deprecation('no_proxy')
```

Comment:
are we sure that `no_proxy` is the one that's intended to be deprecated?  My read was that `skip_proxy` was the one to be deprecated.

Either way, since the ENV variable is `NO_PROXY`, IMO the valid one should be `no_proxy` and the deprecated `skip_proxy`.  They should be the same or it will be more confusing.

Reply:
I actually began the deprecation process for `no_proxy` a little while back because that and the environment variable `NO_PROXY` that the agent also respects mean different things. The env var is a comma separated list of URLs to ignore proxy settings for, but our configuration flag actually semantically meant skip all proxy settings.